crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -557,10 +681,16 @@ impl<P: RoomDataProvider> TimelineController<P> {
             | TimelineFocusKind::Thread { .. } => {
                 return Err(PaginationError::NotSupported);
             }
-            TimelineFocusKind::Event { paginator, .. } => paginator
-                .paginate_backward(num_events.into())
-                .await
-                .map_err(PaginationError::Paginator)?,
+            TimelineFocusKind::Event { paginator, .. } => {
+                if let Some(paginator) = paginator.get() {
```

Comment:
I think I'd do `paginator.get().expect("The paginator must be set at this point")`, and remove the `NotInstantiated` error variant altogether. If a timeline has been successfully built, the paginator instance *will* be set, or the timeline would be in a partially-built state.